gpytorch/lazy/lazy_evaluated_kernel_tensor.py

```diff
@@ -44,12 +44,40 @@ def _getitem(self, row_index, col_index, *batch_indices):
         if self.batch_dims == (0, 2):
             x1 = x1.permute(0, 2, 1).contiguous()
             x1 = x1.view(-1, x1.size(-1), 1)
-        x1 = x1[(*batch_indices, row_index, _noop_index)]
+        try:
+            x1 = x1[(*batch_indices, row_index, _noop_index)]
+        except IndexError:
+            if isinstance(batch_indices, slice):
+                x1 = x1.expand(1, *self.x1.shape[-2:])[(*batch_indices, row_index, _noop_index)]
+            elif isinstance(batch_indices, tuple):
+                if any([not isinstance(bi, slice) for bi in batch_indices]):
+                    raise RuntimeError(
+                        f"Attempting to tensor index a non-batch matrix's batch dimensions. "
+                        "Got batch index {batch_indices} but my shape was {self.shape}"
+                    )
+                x1 = x1.expand(
+                    *([1] * len(batch_indices)),
+                    *self.x1.shape[-2:]
+                )[(*batch_indices, row_index, _noop_index)]
         x2 = self.x2
         if self.batch_dims == (0, 2):
             x2 = x2.permute(0, 2, 1).contiguous()
             x2 = x2.view(-1, x2.size(-1), 1)
-        x2 = x2[(*batch_indices, col_index, _noop_index)]
+        try:
+            x2 = x2[(*batch_indices, col_index, _noop_index)]
+        except IndexError:
```

Comment:
I would add a comment here (which I forgot to do) explaining that the try/catch is for speed purposes.